driver/src/main/com/mongodb/client/MongoCollection.java

```diff
@@ -163,7 +163,7 @@
      * @return a non-null list of distinct values
      * @mongodb.driver.manual reference/command/distinct/ Distinct
      */
-    List<Object> distinct(String fieldName, Object filter, DistinctOptions distinctOptions);
+    MongoIterable<Object> distinct(String fieldName, Object filter, DistinctOptions distinctOptions);
```

Comment:
I'm thinking now we'll need an overload that takes a Class<T> (as with aggregate, etc), so users can control which codec is used to decode each value.